### PR TITLE
arduino-language-server: init at 0.7.1

### PR DESCRIPTION
**Pull request by BattleCh1cken**
###### Description of changes
Packaged the language server for arduino development.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/embedded/arduino/arduino-language-server/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+buildGoModule rec {
+  pname = "arduino-language-server";
+  version = "0.7.1";
+
+  src = fetchFromGitHub {
+    owner = "arduino";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-FnkXtjktj22LCkmFPxzQ72qiVBC1WR/ZiwMsmZPL578=";
```
Comment (sikmir) [inline pkgs/development/embedded/arduino/arduino-language-server/default.nix:12]:
```suggestion
    hash = "sha256-FnkXtjktj22LCkmFPxzQ72qiVBC1WR/ZiwMsmZPL578=";
```

File: pkgs/development/embedded/arduino/arduino-language-server/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  };
+
+  subPackages = [ "." ];
+
+  vendorSha256 = "sha256-Xa26ilo95sQ/6dGvl4gB2bb0vzWXr+WPKLezAnZPeqM=";
```
Comment (sikmir) [inline pkgs/development/embedded/arduino/arduino-language-server/default.nix:17]:
```suggestion
  vendorHash = "sha256-Xa26ilo95sQ/6dGvl4gB2bb0vzWXr+WPKLezAnZPeqM=";
```

File: pkgs/development/embedded/arduino/arduino-language-server/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
+{ lib, stdenv, buildGoModule, fetchFromGitHub}:
+
+
+buildGoModule rec {
+  pname = "arduino-language-server";
+  version = "0.7.1";
+
+  src = fetchFromGitHub {
+    owner = "arduino";
+    repo = pname;
```
Comment (sikmir) [inline pkgs/development/embedded/arduino/arduino-language-server/default.nix:10]: While widely used, `repo` and `pname` semantically are not equal. For example, we can change `pname` of derivation with `overrideAttrs`, but in such case `repo` will be invalid.

File: pkgs/development/embedded/arduino/arduino-language-server/default.nix (hunk 1 above)
Comment (BattleCh1cken) [reply pkgs/development/embedded/arduino/arduino-language-server/default.nix:12]: Is this just because sha256 is already specified by the hash?

Comment (sikmir) [reply pkgs/development/embedded/arduino/arduino-language-server/default.nix]: Look at https://github.com/NixOS/nixpkgs/blob/162b38410d85a3135c436e1e9ed44cb5ee6bd549/pkgs/build-support/fetchurl/default.nix#L60-L69

Comment (BattleCh1cken) [reply pkgs/development/embedded/arduino/arduino-language-server/default.nix]: Aaah. I see.

**Pull request closed without merging**